engine/consensus/matching/core.go

```diff
@@ -141,6 +140,21 @@ func (c *Core) ProcessReceipt(receipt *flow.ExecutionReceipt) error {
 // * error: any error indicates an unexpected problem in the protocol logic. The node's
 //   internal state might be corrupted. Hence, returned errors should be treated as fatal.
 func (c *Core) processReceipt(receipt *flow.ExecutionReceipt) (bool, error) {
+	// setup logger to capture basic information about the receipt
+	log := c.log.With().
+		Hex("receipt_id", logging.Entity(receipt)).
+		Hex("result_id", logging.Entity(receipt.ExecutionResult)).
+		Hex("execution_data_id", receipt.ExecutionResult.ExecutionDataID[:]).
+		Hex("previous_result", receipt.ExecutionResult.PreviousResultID[:]).
+		Hex("block_id", receipt.ExecutionResult.BlockID[:]).
+		Hex("executor_id", receipt.ExecutorID[:]).
+		Logger()
+
+	if c.receipts.HasReceipt(receipt) {
+		log.Debug().Msg("skipping processing of already known receipt")
+		return false, nil
+	}
```

Comment:
Should I put this after the duration metric tracking right below this? Or is this fine?

Reply:
It is fine to be here, I think for that metrics, we are more interested how long a receipt is actually being process.